packages/expo-cli/src/commands/publish/deprecationNotice.ts

```diff
@@ -0,0 +1,9 @@
+import Log from '../../log';
+import { learnMore } from '../utils/TerminalLink';
+
+export async function printDeprecationNotice(): Promise<void> {
+  Log.warn('The last day to use expo publish is 2024-02-12. Migrate to eas update.');
```

Comment:
I think it'd be a good idea to mention the last SDK version that will be supported as well, so that when SDK 50 comes out, people who haven't read the announcement but see this deprecation message aren't confused as to why publishing doesn't work even though it's still 2023.

```suggestion
  Log.warn('The last day to use expo publish is 2024-02-12 and SDK 49 is the last version to support it. Migrate to eas update.');
```